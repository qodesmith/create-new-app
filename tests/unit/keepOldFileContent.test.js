const keepOldFileContent = require('../../modules/keepOldFileContent')
const fs = require('fs-extra')
const path = require('path')
const tempFolder = path.resolve(__dirname, 'temp-keep-old-file-content-test')
const sourcePath = `${tempFolder}/sourceFile.js`
const destinationPathJs = `${tempFolder}/destinationFile.js`
const destinationPathDot = `${tempFolder}/.dotDestinationFile`
const sourceFileContents = "console.log('source file contents')"
const destinationFileContents = "console.log('destination file contents')"
const newContent = '// A comment'

describe('keepOldFileContent', () => {
  // Create temporary folder & files to play with.
  fs.ensureFileSync(sourcePath)
  fs.ensureFileSync(destinationPathJs)
  fs.ensureFileSync(destinationPathDot)

  // Write the contents of the temporary files.
  fs.writeFileSync(sourcePath, sourceFileContents, 'utf8')
  fs.writeFileSync(destinationPathJs, destinationFileContents, 'utf8')
  fs.writeFileSync(destinationPathDot, destinationFileContents, 'utf8')

  // Create content to test against.
  const oldContentsPreservedJs = keepOldFileContent({
    sourcePath,
    destinationPath: destinationPathJs,
  })
  const oldContentsPreservedDot = keepOldFileContent({
    sourcePath,
    destinationPath: destinationPathDot,
  })
  const noContentToOverwriteJs = keepOldFileContent({
    sourcePath,
    destinationPath: './nope.js',
  })
  const noContentToOverwriteDot = keepOldFileContent({
    sourcePath,
    destinationPath: './.dotNope',
  })
  const sourceContentNoOverwriteJs = keepOldFileContent({
    newContent,
    destinationPath: './nope.js',
  })
  const sourceContentNoOverwriteDot = keepOldFileContent({
    newContent,
    destinationPath: './.dotNope',
  })
  const sourceContentWithOverwriteJs = keepOldFileContent({
    newContent,
    destinationPath: destinationPathJs,
  })
  const sourceContentWithOverwriteDot = keepOldFileContent({
    newContent,
    destinationPath: destinationPathDot,
  })

  // Remove all temporary files.
  fs.removeSync(tempFolder)

  it('should copy content from one file and merge it with pre-existing content from the destination file', () => {
    expect(oldContentsPreservedJs).toMatchSnapshot()
    expect(oldContentsPreservedDot).toMatchSnapshot()
  })

  it('should copy content from one file only, when there is no pre-existing file to overwrite', () => {
    expect(noContentToOverwriteJs).toMatchSnapshot()
    expect(noContentToOverwriteDot).toMatchSnapshot()
  })

  it('should write content provided only, when there is no pre-existing file to overwrite', () => {
    expect(sourceContentNoOverwriteJs).toMatchSnapshot()
    expect(sourceContentNoOverwriteDot).toMatchSnapshot()
  })

  it('should write content provided and merge it with pre-existing content from the destination file', () => {
    expect(sourceContentWithOverwriteJs).toMatchSnapshot()
    expect(sourceContentWithOverwriteDot).toMatchSnapshot()
  })
})
