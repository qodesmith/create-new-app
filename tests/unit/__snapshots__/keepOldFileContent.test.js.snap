// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keepOldFileContent should copy content from one file and merge it with pre-existing content from the destination file 1`] = `
"///////////////////////////////////////////////////////////////////
//        The contents of this file have been modified by        //
// create-new-app (https://github.com/qodesmith/create-new-app). //
//     The original contents have been commented out below.      //
///////////////////////////////////////////////////////////////////


console.log('source file contents')



/////////////////////////////
// ORIGINAL CONTENTS BELOW //
/////////////////////////////


// console.log('destination file contents')"
`;

exports[`keepOldFileContent should copy content from one file and merge it with pre-existing content from the destination file 2`] = `
"#################################################################
#        The contents of this file have been modified by        #
# create-new-app (https://github.com/qodesmith/create-new-app). #
#     The original contents have been commented out below.      #
#################################################################


console.log('source file contents')



###########################
# ORIGINAL CONTENTS BELOW #
###########################


# console.log('destination file contents')"
`;

exports[`keepOldFileContent should copy content from one file only, when there is no pre-existing file to overwrite 1`] = `"console.log('source file contents')"`;

exports[`keepOldFileContent should copy content from one file only, when there is no pre-existing file to overwrite 2`] = `"console.log('source file contents')"`;

exports[`keepOldFileContent should write content provided and merge it with pre-existing content from the destination file 1`] = `
"///////////////////////////////////////////////////////////////////
//        The contents of this file have been modified by        //
// create-new-app (https://github.com/qodesmith/create-new-app). //
//     The original contents have been commented out below.      //
///////////////////////////////////////////////////////////////////


// A comment



/////////////////////////////
// ORIGINAL CONTENTS BELOW //
/////////////////////////////


// console.log('destination file contents')"
`;

exports[`keepOldFileContent should write content provided and merge it with pre-existing content from the destination file 2`] = `
"#################################################################
#        The contents of this file have been modified by        #
# create-new-app (https://github.com/qodesmith/create-new-app). #
#     The original contents have been commented out below.      #
#################################################################


// A comment



###########################
# ORIGINAL CONTENTS BELOW #
###########################


# console.log('destination file contents')"
`;

exports[`keepOldFileContent should write content provided only, when there is no pre-existing file to overwrite 1`] = `"// A comment"`;

exports[`keepOldFileContent should write content provided only, when there is no pre-existing file to overwrite 2`] = `"// A comment"`;
